server/topic.go

```diff
@@ -32,22 +36,42 @@ func newTopic(id string) *topic {
 }
 
 // Subscribe subscribes to this topic
-func (t *topic) Subscribe(s subscriber, userID string, cancel func()) int {
+func (t *topic) Subscribe(s subscriber, visitor *visitor, cancel func()) int {
 	t.mu.Lock()
 	defer t.mu.Unlock()
 	subscriberID := rand.Int()
 	t.subscribers[subscriberID] = &topicSubscriber{
-		userID:     userID, // May be empty
+		visitor:    visitor, // May be empty
 		subscriber: s,
 		cancel:     cancel,
 	}
 	return subscriberID
 }
 
+func (t *topic) Stale() bool {
+	return t.getBillee() == nil
+}
+
+func (t *topic) getBillee() *visitor {
+	for _, this_subscriber := range t.subscribers {
+		return this_subscriber.visitor
+	}
+	if t.lastVisitor != nil && t.lastVisitorExpires.After(time.Now()) {
+		t.lastVisitor = nil
+	}
+	return t.lastVisitor
+
+}
```

Comment:
1. Style: Don't use snake_case, use camelCase. And: don't use getters (no getX), use `X` instead, here: `func billee()`
2. This function should use the mutex!
3. This is a strange way of retrieving the billee, IMHO. Why not just keep XXXXXXXXXXXXX aahh toddler. WILL CONTINUE LATER

Reply:
Updated my comment from this afternoon.

Reply:
1 and 2 are done, and I *think* the latest rewrite achieves 3?